flume-cluster.yaml

```diff
@@ -1218,6 +1226,31 @@ Resources:
                 unzip awscliv2.zip
                 ./aws/install
 
+                if [ -e /dev/sdf ]
+                then
+                  mkfs.ext4 /dev/sdf
+                fi
+
+                AVAILABILITY_ZONE=$(curl -s http://169.254.169.254/latest/meta-data/placement/availability-zone)
+                INSTANCE_ID=$(curl -s http://169.254.169.254/2012-01-12/meta-data/instance-id)
+
+                VOLUME_ID=$(aws ec2 describe-volumes --filters Name=tag:SnapshotVolume,Values=${AWS::StackName}-${AWS::Region} Name=status,Values=available Name=availability-zone,Values=$AVAILABILITY_ZONE | jq -r .Volumes[0].VolumeId)
+
+                if [ "$VOLUME_ID" == "null"  ]
+                then
+                  VOLUME_ID=$(aws ec2 create-volume --availability-zone $AVAILABILITY_ZONE --iops 3000  --throughput 125 --size ${FlumeDiskSize} --volume-type gp3 --tag-specifications '{"ResourceType": "volume", "Tags": [{"Key": "SnapshotVolume","Value": "${AWS::StackName}-${AWS::Region}"}, {"Key": "Name", "Value": "${AWS::StackName}-SnapshotVolume"}]}' | jq -r .VolumeId)
+                  while [ "$(aws ec2 describe-volumes --volume-id $VOLUME_ID | jq -r .Volumes[0].State)" != "available" ]
+                  do
+                    sleep 3
+                  done
+                fi
+                aws ec2 attach-volume --device=/dev/sdf --instance-id=$INSTANCE_ID --volume-id=$VOLUME_ID
+
+                while [ "$(aws ec2 describe-volumes --volume-id $VOLUME_ID | jq -r .Volumes[0].State)" != "in-use" ]
+                do
+                  sleep 3
+                done
+
```

Comment:
`mkfs.ext4 /dev/sdf`

You'll want to do this here somehow if you had to create the volume fresh, since it was removed on line 1242

Reply:
Good catch. I've addressed this.